src/ray/raylet/worker_pool.cc

```diff
@@ -1090,6 +1108,12 @@ void WorkerPool::TryKillingIdleWorkers() {
         RAY_CHECK(running_size > 0);
         running_size--;
         rpc::ExitRequest request;
+        if (finished_jobs_.contains(job_id) &&
+            RayConfig::instance().kill_idle_workers_of_terminated_job()) {
+          RAY_LOG(INFO) << "Force exiting worker whose job has exited"
+                        << worker->WorkerId();
+          request.set_force_exit(true);
```

Comment:
Btw, it doesn't need to be done from this PR, but I think we should have more reliable way to force force kill workers than sending a RPC. E.g., send SIGKILL. For example, this issue https://ray-distributed.slack.com/archives/C03H0Q9CUQ3/p1675885877128899?thread_ts=1675460488.542369&cid=C03H0Q9CUQ3 won't be fixed. I will create an issue for this

Reply:
From the user's logs looks like they are calling multi-process from the worker, how do you propose we track those processes?

Reply:
Agree it is orthogonal to this PR